p2p/src/sync/peer.rs

```diff
@@ -132,7 +135,10 @@ where
     }
 
     pub async fn run(&mut self) -> Result<Void> {
+        let mut stalling_interval = tokio::time::interval(*self.p2p_config.sync_stalling_timeout);
+
         self.request_headers().await?;
+        self.last_activity = Some(Instant::now());
```

Comment:
Please use the default `time_getter` here (as it's done in `PeerManager`). Otherwise functional tests may fail unexpectedly.